internal/controllers/clusterrequest_controller.go

```diff
@@ -1172,23 +1179,63 @@ func (t *clusterRequestReconcilerTask) createHwMgrPluginNamespace(
 	return nil
 }
 
+func (t *clusterRequestReconcilerTask) hwMgrPluginNamespaceExists(
+	ctx context.Context, name string) (bool, error) {
+
+	t.logger.InfoContext(
+		ctx,
+		"Plugin: "+fmt.Sprintf("%v", name),
+	)
+
+	namespace := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+	}
+	exists, err := utils.DoesK8SResourceExist(ctx, t.client, name, "", namespace)
+	if err != nil {
+		return false, fmt.Errorf("failed check if namespace exists %s: %w", name, err)
+	}
+
+	return exists, nil
+}
+
 func (t *clusterRequestReconcilerTask) createNodePoolResources(ctx context.Context, nodePool *hwv1alpha1.NodePool) error {
+	chkNodepool := &hwv1alpha1.NodePool{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      nodePool.Name,
+			Namespace: nodePool.Namespace,
+		},
+	}
+	if exists, err := utils.DoesK8SResourceExist(ctx, t.client, nodePool.Name, nodePool.Namespace, chkNodepool); err != nil {
+		return fmt.Errorf("failed check is nodepool exists %s: %w", nodePool.Name, err)
+	} else if exists {
+		return nil
+	}
```

Comment:
We can replace "utils.UPDATE" with an empty string "" in all utils.CreateK8sCR() calls for creating nodePools and namespaces.  If we make this replacement, the check to see if the CR exists will not be needed, as CreateK8sCR() will no longer update the CR if it already exists.

Reply:
Done.